node/core/pvf/execute-worker/src/lib.rs

```diff
@@ -136,11 +138,22 @@ pub fn worker_entrypoint(socket_path: &str, node_version: Option<&str>) {
 			let execute_thread = thread::spawn_worker_thread_with_stack_size(
 				"execute thread",
 				move || {
-					validate_using_artifact(
-						&compiled_artifact_blob,
-						&params,
-						executor_2,
-						cpu_time_start,
+					// Try to enable landlock.
+					#[cfg(target_os = "linux")]
+					let landlock_status = polkadot_node_core_pvf_common::worker::security::landlock::try_restrict_thread()
```

Comment:
So, we have a second path, cases where the kernel supports the landlock API's, but we can't do the restrictions for various reasons, it feels wrong to just log in that case, something unexpected just happened should we stop execution in that case ?

Thinking out loud maybe we should build that config/cmdline right now, something like `ignore_landlock_errors`,  and we should default that on `true` now and then at a later date when everybody is ready we can default that on `false`.  What do you think ?

Reply:
I had a similar thought. Maybe it should be treated as an "internal error" as it is independent of the PVF and candidate - see the "important note" [here](https://github.com/paritytech/polkadot/issues/7041). For now maybe it can remain a warning to make sure it's not common in reality, and then we can upgrade to an internal error later. An error is unlikely enough that it shouldn't affect consensus unless the error happened on some specific kernel version that a lot of validators upgrade to at the same time. The only problem is that I feel like this particular error wouldn't resolve itself, so the validator would not be able to participate in consensus at all - which maybe is a good thing? Will think about it, thanks!

Reply:
Because we will ensure that landlock can be enabled as a requirement for secure-mode (#7073), errors here will be really unlikely. When we have that PR this can be upgraded to an internal error, until then I think we should just log.